content/6_Advanced/SCC.mdx

```diff
@@ -77,7 +77,161 @@ export const problems = {
 
 ## SCCs
 
-<Problems problems={problems.ex} />
+<FocusProblem problem={problems.ex[0]} />
+
+## Solution - Planets and Kingdoms
+
+Just run [Kosaraju's](https://github.com/bqi343/USACO/blob/master/Implementations/content/graphs%20(12)/DFS/SCCK.h) or [Tarjan's](https://github.com/bqi343/USACO/blob/master/Implementations/content/graphs%20(12)/DFS/SCCT.h) SCC algorithm on the graph. 
+
+Then assign each component an $ID$ (starting from $1$).
+
+<Spoiler title="Using Kosaraju's SCC">
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std; 
+/**
+ * Description: Kosaraju's Algorithm, DFS twice to generate 
+ 	* strongly connected components in topological order. $a,b$
+ 	* in same component if both $a\to b$ and $b\to a$ exist.
+ * Time: O(N+M)
+ * Source: Wikipedia
+ * Verification: POI 8 peaceful commission
+*/
+struct SCC {
+	int N; vector<vector<int>> adj, radj;
+	vector<int> todo, comp, comps; vector<bool> vis;
+	void init(int _N) { N = _N; 
+		adj.resize(N), radj.resize(N), comp = vector<int>(N,-1), vis.resize(N); 
+	}
+	void ae(int x, int y) { 
+		adj[x].push_back(y), radj[y].push_back(x); 
+	}
+	void dfs(int x) {
+		vis[x] = 1; for(int y : adj[x]) if (!vis[y]) dfs(y);
+		todo.push_back(x); }
+	void dfs2(int x, int v) {
+		comp[x] = v; 
+		for(int y : radj[x]) if (comp[y] == -1) dfs2(y,v); 
+	}
+	void gen() { // fills allComp
+		for(int i=0; i<N; i++) if (!vis[i]) dfs(i);
+		reverse(begin(todo), end(todo)); 
+		for(int x : todo) if (comp[x] == -1) {
+			dfs2(x,x), comps.push_back(x);
+		}
+	}
+};
+int main() {
+	
```

Comment:
sort of confused why there's space after but not before the start of the function :eyes: